laborator/content/reprezentare-numere/solution/5-odd_even/odd_even.c

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <stdio.h>
+#include <stdlib.h>
+
+void print_binary(int number, int nr_bits)
+{
+	int i;
+	char *bits;
+
+	bits = malloc(sizeof(char) * nr_bits);
+
+	for (i = 0; i < nr_bits; ++i) {
+		*(bits + i) = 1 & number;
+		number >>= 1;
+	}
+
+	printf("0b");
+	for (i = nr_bits - 1; i >= 0; --i)
+		printf("%d", *(bits + i));
+
+	printf("\n");
+
+	free(bits);
+}
+
+void check_parity(int *numbers, int n)
+{
+	int i, curr_nr;
+
+	for (i = 0; i < n; ++i) {
+		curr_nr = *(numbers + i);
+		printf("%5d: ", curr_nr);
+		if (curr_nr & 1)
+			printf("0x%08X\n", curr_nr);
+		else
+			print_binary(curr_nr, 8);
+	}
+}
+
+int main(void)
+{
+	int *numbers, i, *n;
+
+	n = malloc(sizeof(int));
+
+	printf("Size of array: "); scanf("%d", n);
+
+	numbers = malloc(sizeof(int) * (*n));
+
+	for (i = 0; i < *n; ++i) {
+		printf("Number %d: ", i + 1);
+		scanf("%d", numbers + i);
+	}
+
+	check_parity(numbers, *n);
+
+	// free memory
+	free(n);
```

Comment:
```suggestion
	int *numbers, i, 

	printf("Size of array: ");
	scanf("%d", &n);

	numbers = malloc(sizeof(*numbers) * n);

	for (i = 0; i < n; ++i) {
		printf("Number %d: ", i + 1);
		scanf("%d", numbers + i);
	}

	check_parity(numbers, n);

	// free memory
```
Why allocate 4 bytes? Just use a local variable. And ditto about `sizeof(*numbers)`. Also add some error handling in case that `malloc()` fails.

Reply:
I have a Github-related question here. I can directly press a commit button and your modifications will be commited to this PR, but the message would end in the line "Co-authored by: ...", which is not recognized by the `checkpatch` script, and will result in the PR failing the automated tests. This happened before, and I had to squash the commits and force-push. Is there a way I can change the commit message to allow for a "Co-authored by:" statement?

Reply:
Yeah `checkpatch.pl` checks for `Co-developed by:` [1] :upside_down_face:. But you can change the commit message when you apply suggestions. Follow step 4 from here [2].

[1] https://github.com/torvalds/linux/blob/ef5f68cc1f829b492b19cd4df5af4454aa816b93/scripts/checkpatch.pl#L613
[2] https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes